src/world.rs

```diff
@@ -366,6 +366,60 @@ impl Generation {
     }
 }
 
+/// A type implementing `LazyInsert` can be inserted
+/// using `LazyInsertions`.
+pub trait LazyInsert: Send + Sync {
+    /// Inserts the component(s) into the world.
+    fn insert(self, world: &World);
+}
+
+impl<C> LazyInsert for (Entity, C) where C: Component + Send + Sync {
+    fn insert(self, world: &World) {
+        world.write::<C>().insert(self.0, self.1);
+    }
+}
+
+impl<L> LazyInsert for Vec<L>
+    where L:LazyInsert
+{
+    fn insert(self, world: &World) {
+        for item in self {
+            item.insert(world);
+        }
+    }
+}
+
+trait LazyInsertInternal: Send + Sync {
+    fn insert(&mut self, world: &World);
+}
+
+impl<L> LazyInsertInternal for Option<L>
+    where L: LazyInsert
+{
+    fn insert(&mut self, world: &World) {
+        self.take().unwrap().insert(world);
```

Comment:
It's local to the module, so cannot be triggered by the user. I doubt that `expect` or some special casing would be of much use here.